### ISSUE TITLE
keep private key when renewing public device certificate

### ISSUE THREAD

**Issue by reubenmiller** [thin-edge/thin-edge.io#2512]
**Is your feature improvement request related to a problem? Please describe.**

Follow up ticket to https://github.com/thin-edge/thin-edge.io/issues/2489, where the new `tedge cert renew` command was added to renew the public certificate. However in the implementation the private key is recreated. This causes undesirable side-effects, and it is bad practice to re-create the private key as it serves as a "device identity", thus recreating it would change it's identity.

**Describe the solution you'd like**

The private key should not be recreated when renewing the certificate via `tedge cert renew`.

**Describe alternatives you've considered**

**Additional context**

**Comment by rina23q** [comment]
Resolved by #2520 

Note: I did also manual test by using `openssl`. If the both outputs are the same, you can consider the private key and the certificate are the pair.

```sh
openssl pkey -in <KEY_PATH> -pubout -outform pem | sha256sum
openssl x509 -in <CERTIFICATE_PATH> -pubkey -noout -outform pem | sha256sum
```

**Comment by gligorisaev** [comment]
QA has thoroughly checked the feature and here are the results:

- [ ] Test for ticket exists in the test suite.
- [x] QA has tested the function and it's functioning according description.
